File: api/Entity/Property.cs

using TypeGen.Core.TypeAnnotations;

namespace api.Entity;
[ExportTsInterface]
public class Property
{
    public int Id { get; set; }
    public required int PurchasePrice { get; set; }
    public required int MortgageValue { get; set; }
    public required int UpgradeCost { get; set; }
    public required int BoardSpaceId { get; set; }
    public List<PropertyRent> PropertyRents { get; set; } = [];

    //joined GameProperty Id
    public int? GamePropertyId { get; set; }
    public Guid? PlayerId { get; set; }
    public int UpgradeCount { get; set; }
    public bool? Mortgaged { get; set; }
    public Guid? GameId { get; set; }

    //joined From ThemeProperty
    public int? SetNumber { get; set; }
    public string? Color { get; set; }

}

File: ui/src/types/controllers/Property.ts
import { PropertyRent } from "./PropertyRent"

export type Property = {
    id:number
    playerId:string | null
    purchasePrice:number
    mortgageValue:number
    boardSpaceId:number
    upgradeCost:number
    upgradeCount:number
    setNumber:number
    propertyRents:PropertyRent[]
    mortgaged:boolean
    color:string
}

export type PropertyUpdateParams = {
    upgradeCount:number,
    playerId:string
    mortgaged:boolean
}